packages/@vuepress/theme-default/components/PageEdit.vue

```diff
@@ -92,12 +92,24 @@ export default {
         )
       }
 
+      const gitlab = /gitlab.com/
+      if (gitlab.test(docsRepo)) {
```

Comment:
We should probably utilize the `repo` rather than `docsRepo` since there are two configs available to users `repo` and `docsRepo`. Once you update this, we should be good to go!

Reply:
Maybe I still didn't fully understand the concept of `docsRepo`, but I thought the edit links should use that repo instead of the repo of the main site (which is the value of the `repo` variable). If it's not true, just skip the rest of my reply, and I make the change you asked for, I promise! ☺️ 

But if my understanding of `docsRepo` happens to be correct, then there is a very unique use case that my implementation handles well, but the one you suggested does not: let's suppose that you want to
- host your `repo` on GitHub (`"vuejs/vuepress"`)
- but your `docsRepo` on GitLab (`"https://gitlab.com/vuejs/vuepress"`). 

In that case, testing `repo` (`"vuejs/vuepress"`) for the `/gitlab.com/` pattern wouldn't yield a match, so the edit link would follow the GitHub-style (`"https://gitlab.com/vuejs/vuepress/edit/master/docs/guide/using-vue.md"`, note that the `-/` is missing). The reverse is also true (if `repo` is on GitLab, and `docsRepo` is on GitHub).

On the other hand, testing `docsRepo` is always safe, since it either has a value that the user explicitly set, or it falls back to the value of `repo` (see line `56` of this file).

What do you think? I think this implementation handless both the mono-repo-for-the-docs and the default situation pretty well, but if you're not sure about that, then I'll follow your suggestion without further discussion. (Sorry for taking up so much of your time, by the way, you're making an awesome framework, so you deserve some rest!)

Reply:
You are totally right! Thanks for taking the time to explain your rationale behind your implementation.

Given that you have figured out an inefficient path in the code, would it be alright if you updated the BitBuckets implementation to match yours then? You should get full credit for this awesome refactor!

Reply:
Fantastic! :) I'll update the BitBuckets-parts later today (that would've been my follow-up question anyways).